worker/broker.go

```diff
@@ -0,0 +1,510 @@
+type MQTTQuery struct {
+	Pattern string
+	Args    []NamedArgWithType
+}
+
+type BrokerPayload struct {
+	BlockID        int32        `json:"block_id"`
+	BlockIndex     int          `json:"block_index"`
+	ClientID       proto.NodeID `json:"client_id"`
+	ClientSequence uint64       `json:"client_seq"`
+	Events         []MQTTQuery  `json:"events"`
+
+	//Replay API
+	BlockStart int32 `json:"block_start"`
+	IndexStart int   `json:"index_start"`
+	BlockEnd   int32 `json:"block_end"`
+	IndexEnd   int   `json:"index_end"`
+}
+
+type SubscribeEvent struct {
+	ClientID   proto.NodeID
+	DatabaseID proto.DatabaseID
+	ApiName    MQTTAPI
+	Payload    BrokerPayload
+}
+
+type MQTTClient struct {
+	mqtt.Client
+	ListenTopic        string
+	PublishTopicPrefix string
+
+	subscribeEventQueue *gqueue.Queue
+
+	updateCtx    context.Context
+	updateCancel context.CancelFunc
+
+	dbms *DBMS
+}
+
+func convertToMQTTQuery(origins []types.Query) []MQTTQuery {
+	length := len(origins)
+	dests := make([]MQTTQuery, length)
+	for index, origin := range origins {
+		var dest MQTTQuery
+		dest.Pattern = origin.Pattern
+		dest.Args = make([]NamedArgWithType, len(origin.Args))
+
+		for i, fromArg := range origin.Args {
+			var toArg NamedArgWithType
+			toArg.Name = fromArg.Name
+			toArg.Value = fromArg.Value
+
+			switch toArg.Value.(type) {
+			case string:
+				toArg.Type = MQTTString
+			case int64:
+				toArg.Type = MQTTInt64
+			case bool:
+				toArg.Type = MQTTBool
+			case float64:
+				toArg.Type = MQTTFloat64
+			case []byte:
+				toArg.Type = MQTTByte
+			case time.Time:
+				toArg.Value = []byte(fromArg.Value.(time.Time).Format("2006-01-02 15:04:05.999999999-07:00"))
+				toArg.Type = MQTTTime
+			default:
+				toArg.Type = MQTTNull
+			}
+
+			dest.Args[i] = toArg
+		}
+
+		dests[index] = dest
+	}
+	return dests
+}
+
+func convertFromMQTTQuery(origins []MQTTQuery) []types.Query {
+	length := len(origins)
+	dests := make([]types.Query, length)
+	for index, origin := range origins {
+		var dest types.Query
+		dest.Pattern = origin.Pattern
+		dest.Args = make([]types.NamedArg, len(origin.Args))
+
+		for i, fromArg := range origin.Args {
+
+			var toArg types.NamedArg
+			toArg.Name = fromArg.Name
+			switch fromArg.Type {
+			case MQTTString:
+				toArg.Value = fromArg.Value.(string)
+			case MQTTInt64:
+				toArg.Value = fromArg.Value.(int64)
+			case MQTTBool:
+				toArg.Value = fromArg.Value.(bool)
+			case MQTTFloat64:
+				toArg.Value = fromArg.Value.(float64)
+			case MQTTByte:
+				toArg.Value = fromArg.Value.([]byte)
+			case MQTTTime:
+				var err error
+				toArg.Value, err = time.Parse("2006-01-02 15:04:05.999999999-07:00", string(fromArg.Value.([]byte)))
+				if err != nil {
+					toArg.Value = fromArg.Value.([]byte)
+				}
+			default:
+				toArg.Value = fromArg.Value
+			}
+
+			dest.Args[i] = toArg
+		}
+
+		dests[index] = dest
+	}
+	return dests
+}
+
+func NewMQTTClient(config *conf.MQTTBrokerInfo, dbms *DBMS) (c *MQTTClient) {
```

Comment:
**[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
exported function NewMQTTClient should have comment or be unexported